src/builder.rs

```diff
@@ -143,10 +174,23 @@ impl<K: EnrKey> EnrBuilder<K> {
     /// Fails if the identity scheme is not supported, or the record size exceeds `MAX_ENR_SIZE`.
     pub fn build(&mut self, key: &K) -> Result<Enr<K>, EnrError> {
         // add the identity scheme to the content
-        if self.id != "v4" {
+        if self.id != V4 {
             return Err(EnrError::UnsupportedIdentityScheme);
         }
 
+        // If an IP field is set, the NAT fields cannot be set, and vice-versa.
+        if self.content.get(IP_FIELD.as_bytes()).is_some()
+            && self.content.get(NAT_FIELD.as_bytes()).is_some()
+        {
+            return Err(EnrError::NATConfigurationError);
+        }
+
+        if self.content.get(IP6_FIELD.as_bytes()).is_some()
+            && self.content.get(NAT6_FIELD.as_bytes()).is_some()
+        {
+            return Err(EnrError::NATConfigurationError);
+        }
+
```

Comment:
do you think there is the case in which a peer can have a static ip4 field and nat'd ip6? or the other way around ?

Reply:
Yeah. I originally had prevented this, but I decided its was more general to allow these kinds of configurations